crates/ruff_python_parser/src/typing.rs

```diff
@@ -24,23 +24,30 @@ pub enum AnnotationKind {
 
 /// Parse a type annotation from a string.
 pub fn parse_type_annotation(
-    value: &str,
+    // Parsed contents of the string literal node that represents this type annotation.
+    parsed_contents: &str,
+    // Range of the string literal node that represents this type annotation.
     range: TextRange,
+    // The source code.
     source: &str,
 ) -> Result<(Expr, AnnotationKind)> {
     let expression = &source[range];
 
-    if str::raw_contents(expression).is_some_and(|body| body == value) {
+    if str::raw_contents(expression).is_some_and(|raw_contents| raw_contents == parsed_contents) {
         // The annotation is considered "simple" if and only if the raw representation (e.g.,
         // `List[int]` within "List[int]") exactly matches the parsed representation. This
         // isn't the case, e.g., for implicit concatenations, or for annotations that contain
         // escaped quotes.
-        let leading_quote = str::leading_quote(expression).unwrap();
-        let expr = parse_expression_starts_at(value, range.start() + leading_quote.text_len())?;
+        let leading_quote_len = str::leading_quote(expression).unwrap().text_len();
+        let trailing_quote_len = str::trailing_quote(expression).unwrap().text_len();
+        let range = range
```

Comment:
I suspect that this will fail if the string has any prefixes. While uncommon, it is valid python code to e.g. have `r"test"`

Reply:
I think the name of the function is misleading, it does match against both prefixes and quotes:

https://github.com/astral-sh/ruff/blob/c30d34e802ecffc6e820465adc4f7190ede1efb8/crates/ruff_python_ast/src/str.rs#L207-L221